src/caseDiff.js

```diff
@@ -0,0 +1,77 @@
+define([
+    'jquery'
+], function (
+    $
+) {
+    'use strict';
+
+    function compareCaseMappings (baseline, incoming) {
+        // case mappings are a dictionary linking a case property
+        // to a list of questions which populate it.
+        // This function compares the original with the incoming mappings
+        // to produce a diff object
+
+        const additions = {};
+        const deletions = {};
+        const updates = {};
+
+        const allKeys = new Set([...Object.keys(baseline), ...Object.keys(incoming)]);
+
+
+        allKeys.forEach(key => {
+            // If the question is part of the incoming updates, then it is either an addition or an update
+            if (!(key in baseline)) {
+                incoming[key].forEach(update => {
+                    additions[key] = additions[key] || [];
+                    additions[key].push(update);
+                });
+            } else if (key in incoming) {
```

Comment:
Is this condition redundant? If it's not in `baseline` then it _must_ be in `incoming` because `allKeys` contains the union of `baseline` and `incoming` keys.

```suggestion
            } else {
```